util/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java

```diff
@@ -464,6 +461,60 @@ public void subchannelStateIsolation() throws Exception {
     assertThat(pickers.hasNext()).isFalse();
   }
 
+  @Test
+  public void subchannelHealthObserved() throws Exception {
+    // Only the new PF policy observes the new separate listener for health
+    PickFirstLoadBalancerProviderAccessor.setEnableNewPickFirst(true);
+    // PickFirst does most of this work. If the test fails, check IS_PETIOLE_POLICY
+    Map<Subchannel, LoadBalancer.SubchannelStateListener> healthListeners = new HashMap<>();
+    loadBalancer = new RoundRobinLoadBalancer(new ForwardingLoadBalancerHelper() {
+      @Override
+      public Subchannel createSubchannel(CreateSubchannelArgs args) {
+        Subchannel subchannel = super.createSubchannel(args.toBuilder()
```

Comment:
Why not add a check here that args contains the IS_PETIOLE_POLICY?

Reply:
Because the subchannel won't have that arg. It is in the ResolvedAddresses's Attributes, not the EAG or CreateSubchannelArgs.